### ISSUE TITLE
Feature Request: More options for subtasks

### ISSUE THREAD

**Issue by ikomhoog** [wekan/wekan#3626]
I see Subtasks as a thing that needs to be done before this task is done.
Currently, it's just a relation and there is no feedback on how many subtasks are already "Done" like for checklists.
Some of my tasks have the same subtasks, and having multiple parents is currently not supported
that is why I'm requesting these 3 things:

1. I would like a small indicator of done subtasks/total subtasks like currently for the checklists.
There needs to be a setting as to what list is marking them as "Done"

2. allow tasks to have multiple parents.
For example, I have cards ABC and A needs to be done before B and C can start, I would assign A as a subtask to both B and C because it needs to be done first.

3. I would also like to add subtasks via the same menu where you now create new ones.
So instead of always needing to go to the child and assign a parent(or multiple after this feature), you can just search it from the same field.

Do these things look viable with the current code?

**Comment by xet7** [comment]
@ikomhoog 

Do you mean like currently: Checklist name / Checklist Item, but with unlimited hierarchy of subsubtasks etc?

Or what it should look like?

[Current way how subtasks work](https://github.com/wekan/wekan/wiki/Subtasks) with separate boards is problematic, I don't know would it be possible to migrate old subtasks in any way to that new hierarchy of subtasks.

Also, sometime in future card layout will change to fullscreen like this https://github.com/wekan/wekan/issues/1574#issuecomment-609095074 that will give more space to view card content.

I have not looked how visually subtasks are done in other kanban software. Someone could add some screenshots or screenshot mockups about what it should look like. For example, here is how we did think various alternatives about how swimlanes should work #955 .

**Comment by ikomhoog** [comment]
"Do you mean like currently: Checklist name / Checklist Item, but with unlimited hierarchy of subsubtasks etc?" 
Yes but i would only show this cards subtasks and not subsubtasks for performance reasons.

"Current way how subtasks work with separate boards is problematic, I don't know would it be possible to migrate old subtasks in any way to that new hierarchy of subtasks."
When I look at the current code I think the linkage should be a seperate object.
Currently the cards have a parentId variables.
The connectionObject would have 2 variables parentId and relationId.
Then for finding the current subtasks you can search for connectionObjects that have the currentCardId as parentId.
And when checking for parent prefix you can search for connectionObjects that have the currentCardId as relationId.
this would allow for multiple parents to be assigned to a card.

As for the visuals, I altered the checklist badge in CSS with a different icon("af-list-alt") to make it look like a card and use it for a subtask progress badge.
I also added a second line to the prefix of the child card to represent 2 depentant tasks.
And I added a dropdown which is to select what list will mark the subtasks as "done" so the badge will work.
![image](https://user-images.githubusercontent.com/9083045/109949859-2c21e400-7cdc-11eb-96b1-bdd9e9223ccc.png)

Here I copied the dropdown from  "Change cards parents" into the subtasks screen (in paint, no coding).
This way we can select currently existing tasks as subtasks.
The current way of "Change cards parents" would be disabled because it doesn't have a neat way of showing multiple connections(yet), maybe when the card layout gets to fullscreen it can be expanded to show all connections.
Whereas the subtask div does already scale with multiple connections.
![image](https://user-images.githubusercontent.com/9083045/109949872-304e0180-7cdc-11eb-8b00-5ab5d7cf46a5.png)

I just downloaded the repo and will look around to see if I can get some working prototypes.

**Comment by benjaminhrivera** [comment]
I second these feature addition / changes. They make viewing cards easier for users and to what they are and make it easier to child tasks to other.

**Comment by zhoujunhaohao** [comment]
YES

**Comment by avatar1024** [comment]
I also agree, these features would be great. Especially the possibility of assign a current card as a subtask of another card in parent board.
